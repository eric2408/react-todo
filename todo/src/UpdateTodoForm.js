import React, {useState} from "react";


function UpdateTodoForm({update, id, tasks}) {
    const [task, setTask] = useState(tasks);
    const handleChange = (e) => {
        setTask(e.target.value);
    };
    const handleSubmit = (e) => {
        e.preventDefault();
        if (task !== ''){
            update( id, task);
        }
        setTask("");
    };


    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="task">Update Todo: </label>
            <input id="task" name="task" type="text" value={task} onChange={handleChange}/>
            <button>Update Todo</button>
        </form>
    );
}

export default UpdateTodoForm;