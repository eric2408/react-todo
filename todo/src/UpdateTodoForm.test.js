import React from "react";
import { render} from "@testing-library/react";
import UpdateTodoForm from "./UpdateTodoForm";

it("renders without crashing", function() {
    render(<UpdateTodoForm />);
  });
  
it("matches snapshot", function() {
    const { asFragment } = render(<UpdateTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});