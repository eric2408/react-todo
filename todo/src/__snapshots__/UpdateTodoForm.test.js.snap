// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Update Todo: 
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button>
      Update Todo
    </button>
  </form>
</DocumentFragment>
`;
